pkg/catalog/catalog.go

```diff
@@ -37,12 +37,12 @@ func (c *Catalog) Search(name string) []profilesv1.ProfileDescription {
 }
 
 // Get returns the profile description `profileName`.
-func (c *Catalog) Get(catalogName, profileName string) profilesv1.ProfileDescription {
+func (c *Catalog) Get(catalogName, profileName string) (bool, profilesv1.ProfileDescription) {
 	for _, p := range c.profiles {
```

Comment:
how about returning `*profilesv1.ProfileDescription` which means you can return `nil` if it doesn't exist?

Reply:
I knew you would point this out! :laughing: I didn't want to return a pointer, because then you can modify whats in the inmemory cache?

Reply:
that's really an implementation detail of the fact that it's an inmemory cache?

You could do a DeepCopy on it, and return a pointer to that?